packages/core/src/payment/strategies/mollie/mollie-payment-strategy.ts

```diff
@@ -85,30 +87,36 @@ export default class MolliePaymentStrategy implements PaymentStrategy {
             this._mountElements();
         }
 
-        if (methodsNotAllowedWhenDigitalOrder.includes(methodId)) {
-            const cart = state.cart.getCartOrThrow();
-            const cartDigitalItems = cart.lineItems.digitalItems;
+        this._unsubscribe = this._store.subscribe(
+            async state => {
+                if (state.paymentStrategies.isInitialized(methodId)) {
 
-            if (cartDigitalItems && cartDigitalItems.length > 0) {
-                const { containerId } = this._getInitializeOptions();
+                    if (methodId && gatewayId && mollie) {
```

Comment:
Do we need to do this check again? Since we have already done this potentially at line 52?